### ISSUE TITLE
Annotated Drops Metadata_Well_Position column when the Platemap and Profile dataset use the same name

### ISSUE THREAD

**Issue by kenibrewer** [cytomining/pycytominer#240]
After running pycytominer.annotate on a platemap and profiles_df that already had matching well_position column names, I noticed that the Metadata_Well column disappeared. 

I tracked it down to this line of code: https://github.com/cytomining/pycytominer/blob/8e5d27f592b10b10a73c55b6335e30e1c6c492fd/pycytominer/annotate.py#L82

This doesn't take into account the possibility that the two columns might be the same when dropping one. I'll submit a pull request implementing this fix.

**Comment by kenibrewer** [comment]
Below is an example code fragment modified from the test/test_annotate.py file that illustrates this problem. This code fails if the PLATEMAP_WELL_COLUMN is set to 'Metadata_Well'. 

```{python}
import pathlib
import random
import tempfile

import pandas as pd
from pycytominer.annotate import annotate

random.seed(123)

# Get temporary directory
TMPDIR = tempfile.gettempdir()

# Setup a testing file
OUTPUT_FILE = pathlib.Path(f"{TMPDIR}/test.csv")

PLATEMAP_WELL_COLUMN = "Metadata_well_position"

# Build data to use in tests
DATA_DF = pd.concat(
    [
        pd.DataFrame(
            {"Metadata_Well": ["A01", "A02", "A03"], "x": [1, 3, 8], "y": [5, 3, 1]}
        ),
        pd.DataFrame(
            {"Metadata_Well": ["B01", "B02", "B03"], "x": [1, 3, 5], "y": [8, 3, 1]}
        ),
    ]
).reset_index(drop=True)

PLATEMAP_DF = pd.DataFrame(
    {
        PLATEMAP_WELL_COLUMN: ["A01", "A02", "A03", "B01", "B02", "B03"],
        "gene": ["x", "y", "z"] * 2,
    }
).reset_index(drop=True)


def test_annotate():

    # create expected result prior to annotate to distinguish modifications
    # performed by annotate to provided dataframes.
    expected_result = (
        PLATEMAP_DF.merge(DATA_DF, left_on=PLATEMAP_WELL_COLUMN, right_on="Metadata_Well")
        .rename(columns={"gene": "Metadata_gene"})
        .drop(PLATEMAP_WELL_COLUMN, axis="columns")
    )

    result = annotate(
        profiles=DATA_DF,
        platemap=PLATEMAP_DF,
        join_on=[PLATEMAP_WELL_COLUMN, "Metadata_Well"],
    )

    print(result)
    pd.testing.assert_frame_equal(result, expected_result)
```

**Comment by kenibrewer** [comment]
Resolved by #241
